MLS.Agent/DotnetKernelJupyterInstaller.cs

```diff
@@ -0,0 +1,111 @@
+                }
+            }
+            catch (KernelInstallationFailureException e)
+            {
+                console.Out.WriteLine($".NET Kernel Installation failed with error {string.Join("\n", e.Message)}");
+                return -1;
+            }
+        }
+
+        private static void Installkernel(IEnumerable<DirectoryInfo> dataDirectories, IConsole console)
+        {
+            foreach(var directory in dataDirectories)
+            {
+                if (directory.Exists)
+                {
+                    var kernelDirectory = directory.Subdirectory("kernels");
+                    if(!kernelDirectory.Exists)
+                    {
+                        kernelDirectory.Create();
+                    }
+
+                    var dotnetkernelDir = kernelDirectory.Subdirectory(".NET");
+                    if(!dotnetkernelDir.Exists)
+                    {
+                        dotnetkernelDir.Create();
+                    }
+
+                    console.Out.WriteLine($"Installing the .NET kernel in directory: {dotnetkernelDir.FullName}");
+                    
+                    // Copy the files into the kernels directory
+                    File.Copy("kernels.json", Path.Combine(dotnetkernelDir.FullName, "kernels.json"));
+                    File.Copy("logo-32x32.png", Path.Combine(dotnetkernelDir.FullName, "kernels.json"));
+                    File.Copy("logo-64x64.png", Path.Combine(dotnetkernelDir.FullName, "kernels.json"));
+                    console.Out.WriteLine($"Finished installing the .NET kernel in directory: {dotnetkernelDir.FullName}");
+                }
+            }
+        }
+
+        public static async Task<IEnumerable<DirectoryInfo>> GetJupyterDataPaths(IReadOnlyCollection<string> jupyterPathInfo)
+        {
+            var indexOfDataNode = Array.FindIndex(jupyterPathInfo.ToArray(), element => element.TrimLineEndings().CompareTo("data:") == 0);
+            if (indexOfDataNode != -1)
+            {
+                var currentPaths = new List<string>();
+                for (var index = indexOfDataNode + 1; index < jupyterPathInfo.Count; index++)
+                {
+                    string element = jupyterPathInfo.ElementAt(index);
+                    if (!string.IsNullOrWhiteSpace(element))
+                    {
+                        if (element.Trim('\r', '\n').EndsWith(":"))
+                        {
+                            break;
+                        }
+                        else
+                        {
+                            currentPaths.Add(element);
+                        }
+                    }
+                }
+
+                return currentPaths.Select(dir => new DirectoryInfo(dir));
+            }
+            else
+            {
+                throw new KernelInstallationFailureException($"Could not find the jupyter kernel installation directory." +
```

Comment:
Rather than throw here, why not just return an empty collection and let the caller determine what action to take?